Tests/StreamChatUITests/AssertSnapshot.swift

```diff
@@ -0,0 +1,71 @@
+//
+// Copyright © 2021 Stream.io Inc. All rights reserved.
+//
+
+import SnapshotTesting
+import XCTest
+
+/// Snapshot of a view controller. All variants will be tested by default. For each variant, it will take a snapshot.
+/// It is recommended to test all trait variants only for view controllers, and for small view components only test
+/// the default variant. This is to minimize the amount of pictures stored in the repo.
+///
+/// - Parameters:
+///   - vc: The view controller to be tested.
+///   - isEmbededInNavigationController: If the view controller should be embeded in a navigation controller.
+///   Useful to test the navigation bar.
+///   - variants: The variants that a snapshot will be taken. All variants should be tested for a view controller.
+///   - record: True if a new reference should be saved. False by default,
+///   so that the newly captured snapshot is compared with the current reference.
+func AssertSnapshot(
+    _ vc: UIViewController,
+    isEmbededInNavigationController: Bool = false,
+    variants: [SnapshotVariation] = SnapshotVariation.allCases,
+    record: Bool = false,
+    line: UInt = #line,
+    file: StaticString = #file,
+    function: String = #function
+) {
+    let viewController = isEmbededInNavigationController ? UINavigationController(rootViewController: vc) : vc
+    variants.forEach { variant in
+        assertSnapshot(
+            matching: viewController,
+            as: .image(traits: variant.trait),
+            named: variant.rawValue,
+            record: record,
+            file: file,
+            testName: function,
+            line: line
+        )
+    }
+}
+
+/// Snapshot of a view. Used for view components. Only the default variant is tested.
+/// To minimize the amount of snapshot pictures saved in the repo, the rest of the variants should
+/// be tested in view controllers.
+///
+/// - Parameters:
+///   - view: The component to be tested.
+///   - variants: The variants that a snapshot will be taken.
+///   Only the light and dark variants should be tested in small components.
+///   - record: True if a new reference should be saved. False by default,
+///   so that the newly captured snapshot is compared with the current reference.
+func AssertSnapshot(
+    _ view: UIView,
+    variants: [SnapshotVariation] = [.light, .dark],
```

Comment:
why just these two variants?

Reply:
My initial idea was that for small components, we would not care about the font sizes, because that should be tested in view controllers. But thinking about it better now, that is probably a bad assumption 🤔 In the message cells, we might need to test the font sizes without them being in a view controller. So probably is better to switch back these to all cases. I guess we could also test each message cell, in a view controller with one cell, not sure... I think we need to explore and see what works best for us.

Reply:
Yeah, we'll see once we start using it for real.